php/wordcount.php

```diff
@@ -7,37 +7,30 @@
 
 $array = array();
 while (false !== ($line = fgets($stdin))) {
-  $words = preg_split('/\s+/', $line);
+    $words = preg_split('/\s+/', $line, -1, PREG_SPLIT_NO_EMPTY);
     foreach($words as $word){
-        if(empty($word) &&  $word !== "0"){
-            continue;
-        }
-        if (!array_key_exists($word,$array)){
+        if (isset($array[$word])){
+            $array[$word]++;
+        } else {
             $array[$word] = 1;
-            continue;
         }
-        $array[$word]++;
     }
-
-
 }
 fclose($stdin);
 $array2 = array();
-
 foreach($array as $key => $value){
-
-    if (!array_key_exists($value,$array2)){
+    if (!isset($array2[$value])){
```

Comment:
This if statement is redundant. This entire loop can be simplified to:

    foreach ($array as $key => $value) {
        $array2[$value][] = $key;
    }